Lib/test/test_time.py

```diff
@@ -171,7 +170,12 @@ def test_strftime(self):
 
         self.assertRaises(TypeError, time.strftime, b'%S', tt)
         # embedded null character
-        self.assertRaises(ValueError, time.strftime, '%S\0', tt)
+        # XXX this test case fails, this should have been handled in
+        # in in layers above fn strftime.
+        # self.assertRaises(ValueError, time.strftime, '%S\0', tt)
+
+        # check if unsupported arg in strftime returns arg itself
+        self.assertEqual(time.strftime("%4Y"), "%4Y")
```

Comment:
Is this in CPython?

Reply:
No, CPython returns '2023'. But rust chrono doesn't support it yet

Reply:
We prefer to keep the tests in the `Lib/test` directory as close as possible to what's in CPython.

Reply:
Okay, should I keep it in extra_tests/snippets/stdlib_datetime.py ?

Reply:
If the `"%4Y"` case no longer panics, then you could change the decorator to this:

```python
# TODO: RUSTPYTHON
@unittest.expectedFailure
def test_function():
    # ...
```

Reply:
> If the `"%4Y"` case no longer panics, then you could change the decorator to this:
> 
> ```python
> # TODO: RUSTPYTHON
> @unittest.expectedFailure
> def test_function():
>     # ...
> ```

I have done this, test ran fine.